src/DynamoCore/Models/DynamoModel.cs

```diff
@@ -2265,9 +2265,44 @@ public void AddWorkspace(WorkspaceModel workspace)
             };
 
             _workspaces.Add(workspace);
+            //TODO move this to some handler for the event...
+            CheckForXMLDummyNodes(workspace);
             OnWorkspaceAdded(workspace);
         }
 
+        private void CheckForXMLDummyNodes(WorkspaceModel workspace)
+        {
+            //if the graph that is opened contains xml dummynodes log a notification 
+            if (workspace.containsXmlDummyNodes())
+            {
+                this.Logger.LogNotification("DynamoViewModel",
+                  Resources.UnresolvedNodesWarningTitle,
+                  Resources.UnresolvedNodesWarningShortMessage,
+                  Resources.UnresolvedNodesWarningMessage);
+
+                //raise a window as well so the user is clearly alerted to this state.
+                DisplayXmlDummyNodeWarning();
+            }
+        }
+        private void DisplayXmlDummyNodeWarning()
+        {
+           var xmlDummyNodeCount = this.CurrentWorkspace.Nodes.OfType<DummyNode>().
+                Where(node => node.OriginalNodeContent is XmlElement).Count();
+
+           Logging.Analytics.LogPiiInfo("XmlDummyNodeWarning",
+               xmlDummyNodeCount.ToString());
+
+            string summary = Resources.UnresolvedNodesWarningShortMessage;
+            var description = Resources.UnresolvedNodesWarningMessage;
+            const string imageUri = "/DynamoCoreWpf;component/UI/Images/task_dialog_future_file.png";
+            var args = new TaskDialogEventArgs(
+               new Uri(imageUri, UriKind.Relative),
+               Resources.UnresolvedNodesWarningTitle, summary, description);
+
+            args.AddRightAlignedButton((int)ButtonId.Proceed, Resources.OKButton);
+            OnRequestTaskDialog(null, args);
```

Comment:
I thought we are only going to reply on Notification Center but not showing any dialog at all?

Reply:
@QilongTang that was for Colin's task we discussed this, we did not discuss here and @Racel has not replied yet to my question above.